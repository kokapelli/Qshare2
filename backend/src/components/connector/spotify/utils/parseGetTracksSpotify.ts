import { QShareTrack, QShareArtist } from "../../../music/musicTypes";

interface SpotifyTrack {
    name: string;
    uri: string;
    artists: SpotifyArtist[];
    album: SpotifyAlbum;
    popularity: number;
}

interface SpotifyArtist {
    uri: string;
    name: string;
}

interface SpotifyAlbum {
    images: SpotifyAlbumImage[];
}

interface SpotifyAlbumImage {
    width: number;
    height: number;
    url: string;
}

interface SpotifyGetTracksResponse {
    tracks: {
        name: string
        items: SpotifyTrack[]
        limit: number
    };
}

export interface QShareGetTrackResponse extends QShareTrack {
    popularity: number;
}

export interface QShareGetTracksResponse {
    limit: number;
    tracks: QShareGetTrackResponse[];
}

export const parseGetTracksSpotify = (data: SpotifyGetTracksResponse) => {
    const response: QShareGetTracksResponse = {} as any;
    const tracks: QShareGetTrackResponse[] = [];

    response.tracks = tracks;
    response.limit = data.tracks.limit;
    for (const track of Array.from(data.tracks.items)) {
        const newTrack: QShareGetTrackResponse = {} as any;
        newTrack.name = track.name;

        const newArtists: QShareArtist[] = [];
        for (const artist of track.artists) {
            const newArtist: SpotifyArtist = {
                uri: artist.uri,
                name: artist.name
            };

            newArtists.push(newArtist);
        }

        newTrack.artists = newArtists;

        const newImages = [];
        for (const image of track.album.images) {
            const newImage: SpotifyAlbumImage = {
                height: image.height,
                width: image.width,
                url: image.url
            };
            newImages.push(newImage);
        }

        newTrack.images = newImages;

        newTrack.uri = track.uri;
        newTrack.popularity = track.popularity;
        tracks.push(newTrack);
    }

    return response;
};
