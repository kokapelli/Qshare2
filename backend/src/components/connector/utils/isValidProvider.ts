import { SPOTIFY_PROVIDER } from "../../constants/providers";

const isValidProvider = (provider: string) => {
    return provider === SPOTIFY_PROVIDER;
};

export { isValidProvider };