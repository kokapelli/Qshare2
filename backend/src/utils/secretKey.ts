import crypto from "crypto";

const SECRET_KEY = crypto.randomBytes(256);

export default SECRET_KEY;
