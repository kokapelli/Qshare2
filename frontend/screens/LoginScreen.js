import React, { Component } from 'react'
import { connect } from 'react-redux'
import { StyleSheet, View, Text, TouchableOpacity} from 'react-native'
import { login } from '../actions/login'

export class LoginScreen extends Component {
  static navigationOptions = {
    header: null
  }

  componentDidMount () {
    this.props.login()
  }

  componentWillReceiveProps(newProps){
    if (newProps.redirect && newProps.success){
      this.props.navigation.navigate('CreateRoom');
    }
  }

  render () {
    const {loading, error} = this.props;
    if (loading){
      return (
        <View style={styles.container}>
          <Text testID='loading' style={styles.buttonText}>Loading...</Text>
        </View>)
    }

    if (error){
      return (
        <View style={styles.container}>
          <Text testID='errorText' style={styles.buttonText}>Something went wrong</Text>
          <TouchableOpacity onPress={() => this.props.navigation.navigate('Main')}>
            <Text testID='errorBtn' style={styles.buttonText}>Go back!</Text>
          </TouchableOpacity>
        </View>)
    }

    return (
      <View style={styles.container}/>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#181818'
  },
  headerContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  headerText: {
    fontSize: 80,
    textAlign: 'center',
    margin: 10,
    color: 'white'
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center'
  },
  button: {
    borderWidth: 1,
    margin: 5,
    borderColor: '#999999',
    alignItems: 'center',
    justifyContent: 'center',
    width: 250,
    height: 70,
    backgroundColor: '#1db954',
    borderRadius: 100
  },
  buttonText: {
    color: 'white',
    fontSize: 20
  },
  founderContainer: {
    alignItems: 'center',
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    paddingBottom: 10
  },
  founderText: {
    color: 'white',
    fontSize: 15
  }
})

const mapStateToProps = (state) => ({
  loading: state.login.loading,
  redirect: state.login.redirect,
  success: state.login.success,
  error: state.login.error
})

export default connect(mapStateToProps, {login})(LoginScreen)
