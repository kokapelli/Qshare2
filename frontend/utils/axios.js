import Axios from "axios"
import { settings } from "../settings/settings";

const axios = Axios.create()
axios.defaults.headers.common['Provider'] = settings.provider

export default axios