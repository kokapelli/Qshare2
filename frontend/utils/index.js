export const findByTestAttr = (component, attr)  => {
    const wrapper = component.find(`[testID='${attr}']`);
    return wrapper
  }
  