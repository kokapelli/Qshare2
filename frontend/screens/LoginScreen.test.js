import { LoginScreen } from './LoginScreen'
import { shallow, configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { findByTestAttr } from '../utils/'

const login = jest.fn()
configure({ adapter: new Adapter() });

const initialState = {
    loading: false,
    success: false,
    redirect: false,
  };

//TODO Test Navigation
describe('LoginScreen Snapshot', () => {

    it('matches the snapshot', () => {
        const tree = renderer.create(
        <LoginScreen login={login} />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('LoginScreen Units', () => {

    beforeEach(()=>{
       this.component = shallow(<LoginScreen login={login}/>)
    })

    it('Component renders properly', () => {
        const rendered = renderer.create(<LoginScreen login={login}/>).toJSON()
        expect(rendered).toBeTruthy()
    });

    it('Loading... should be displayed when loading is true', () => {
        this.component.setProps({ ...initialState, loading:true });
        const loading = findByTestAttr(this.component, 'loading')
        expect(loading.props().children).toEqual("Loading...")
    });

    it('An empty view should be displayed when loading is false', () => {
        expect(this.component).toEqual({})
    });
})