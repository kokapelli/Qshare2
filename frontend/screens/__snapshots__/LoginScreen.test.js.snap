// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Snapshot matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#181818",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
